projects/extension/src/background/ConnectionManager.ts

```diff
@@ -187,14 +192,21 @@ export class ConnectionManager extends (EventEmitter as { new(): StateEmitter })
    * @param spec - ChainSpec of chain to be added
    * @param jsonRpcCallback - The jsonRpcCallback function that should be triggered
    * @param relayChainName - optional string when parachain is added to depict the relay chain name
+   * 
+   * @returns addedChObj - An object that contains the chain info and the healthchecker for the connected chain
    */
-  async addChain (name: string, chainSpec: string, jsonRpcCallback: smoldot.SmoldotJsonRpcCallback, relayChainName?: string): Promise<smoldot.SmoldotChain> {
+  async addChain (
+    name: string,
+    chainSpec: string,
+    jsonRpcCallback: SmoldotJsonRpcCallback,
+    relayChainName?: string): Promise<ChainInstance> {
     if (!this.#client) {
       throw new Error('Smoldot client does not exist.');
     }
     let relay: Network | undefined = undefined;
     let addChainOptions = {} as SmoldotAddChainOptions;
 
+    const healthChecker = await (smoldot as any).healthChecker();
```

Comment:
This isn't removed

Reply:
This is the init of the healthChecker that happens for each chain (1 chain = 1 healthChecker). Why would I remove it?
Its initialized here and then appMediator is using it in order to setup the setJsonRpc, start the healthchecker and stop it as needed